tests/unit_tests/test_spectrum.py

```
@@ -135,53 +120,57 @@ def test_init_applies_frame_cutoff_before_calculating_r0(tmp_path):
+    """Test that a missing input path is rejected."""
     with pytest.raises(ValueError, match="does not appear to be a file"):
         Spectrum(tmp_path / "missing.npy")
 
 
 def test_init_rejects_non_npy_file(tmp_path):
-    """Test that constructing Spectrum from an existing file raises ValueError when the suffix is not .npy."""
+    """Test that file inputs must be .npy files."""
     infile = tmp_path / "edges.txt"
     infile.write_text("not an npy file")
-
     with pytest.raises(ValueError, match="must end in .npy"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)